lightning/src/ln/outbound_payment.rs

```diff
@@ -684,13 +687,15 @@ pub(super) struct SendAlongPathArgs<'a> {
 
 pub(super) struct OutboundPayments {
 	pub(super) pending_outbound_payments: Mutex<HashMap<PaymentId, PendingOutboundPayment>>,
+	pub(super) awaiting_invoice_flag: AtomicBool,
 	pub(super) retry_lock: Mutex<()>,
 }
 
 impl OutboundPayments {
 	pub(super) fn new() -> Self {
 		Self {
 			pending_outbound_payments: Mutex::new(new_hash_map()),
+			awaiting_invoice_flag: AtomicBool::new(false),
 			retry_lock: Mutex::new(()),
 		}
 	}
```

Comment:
Let's refactor this such that the constructor takes a `HashMap` and the members other than `pending_outbound_payments` are private. Then you can drop `set_awaiting_invoice_flag` and move that logic inside the constructor to determine the initial value of `awaiting_invoice_flag`.

Reply:
Sounds like a great idea to make the flag, and lock private!
Refactor the code in [fa5cf60](https://github.com/lightningdevkit/rust-lightning/commit/fa5cf6003f56940682f578688cd33c68838086e9)
Thanks!